### PR TITLE
Fix command argument splitting to parse quoted arguments

### LINKED ISSUE #206: Improve error handling of `klog edit`
I'm running a Windows 11 machine and I'm trying to get klog to open my files in vscode, but the files will just open in notepad. Klog doesn't "fail" to open the file, just not in the correct editor, and do not produce any output for me to debug.

- I have verified the paths is correct.
- I have tried to make the environment variable a string.
- I have tried using forward slashes.
- I have tried using a symlink pointing to code.exe.

```powershell
> $env:EDITOR="C:\Program Files\Microsoft VS Code\Code.exe"
> $env:EDITOR
C:\Program Files\Microsoft VS Code\Code.exe
> klog edit "C:\Users\elias\Desktop\time_records\xxx.klg"
```

### PR DESCRIPTION
**Pull request by tscpp**
I wrote a simple parse function to split arguments. The parser allows multiple spaces between arguments and can parse quoted arguments. The function also removed the quotes from the first argument as it is expected to be the executable.

I also made it so `tryCommands` logs all errors from failed commands.

Tell me if there anything I can fix or improve in this pull request. Otherwise, the pull request is ready to be merged (if the changes are wanted).

Fixes #206

### CODE REVIEW

**Comment by jotaen** [comment]
Thanks for submitting this! It might take a while for me to look into it, I’m pretty busy right now… (just so that you know 🙂)

**Comment by jotaen** [comment]
While testing your branch, I’ve realised that my original implementation had wrong premises to begin with, and so I tried to sort that out properly. Unfortunately, the parsing method that you came up with became obsolete that way, so sorry about that…

Instead of storing the pre-configured commands as single string, they should actually be stored as array of strings ([see diff](https://github.com/jotaen/klog/pull/207/commits/d3e5f3071c2dda724a423a7fa523f4a14018093e#diff-45c90eba109ed5e056ff0bfdb1d90d1468a5ff663b87b40f607af9f34572adedR13)), where command and arguments are separated from each other. That way, it’s always unambiguous what the command is, and what the arguments are. The same applies for `$EDITOR`, which then is correctly interpreted as atomic path, regardless of any spaces that it might contain.

I also improved the error handling, so when `$EDITOR` is set but fails to open, then `klog edit` will fail right away, and also print more helpful details about the error. I think that’s better than just silently failing over to the built-in default commands, as it avoids potential confusion about what was going on (like in your case).

```
$ klog edit myfile.klg
Error: Cannot open preferred editor
$EDITOR variable was: C:\Program Files\Some Non-Existing Editor
```

So basically, when `$EDITOR` is set, it must be correct. Only if it’s not set, klog will try the built-in editors one after the other.
